### ISSUE TITLE
`resolveId` doesn’t seem to work in webpack

### ISSUE THREAD

**Issue by Codpoe** [unjs/unplugin#19]
I use `resolveId` and `load` to implement the plugin logic:
```js
export default createUnplugin(() => ({
  name: 'unplugin-starter',
  resolveId(id) {
    return id === 'my-unplugin-virtual-id' ? id : null
  },
  load(id) {
    if (id === 'my-unplugin-virtual-id')
      return 'export default \'hello my-unplugin\''
  },
}))
```

When I used this plugin in webpack, it reported an error. But it's ok in vite.
```js
import str from 'my-unplugin-virtual-id'
```

```
Module not found: Error: Can't resolve 'my-unplugin-virtual-id' in 'D:\Projects\my-unplugin\playground\webpack\src'
```

> This is my minimal demo that can reproduce the problem：https://github.com/Codpoe/unplugin-minimal-demo.
And I had this problem on Windows, I haven’t tested it on Mac...

**Comment by TropicalRaisel** [comment]
When importing, you need to do something like `my-unplugin-virtual-id/{framework}`, where `framework` is either `webpack`, `rollup`, or `vite`.

Just out of curiosity, why are you using the `load` statement like that? It seems unnecessary.

**Comment by Codpoe** [comment]
> When importing, you need to do something like my-unplugin-virtual-id/{framework}, where framework is either webpack, rollup, or vite.

Sorry, I don't understand what you mean. I think my plugin has been set up correctly:
```js
module.exports = {
  entry: './src/index.js',
  plugins: [
    require('my-unplugin/webpack').default(),
  ],
}
```

---

> why are you using the load statement like that? It seems unnecessary.

I’m just giving a simplest example.😂

**Comment by TropicalRaisel** [comment]
I'm saying that your "import str" statement is strange b/c:
1. There's no "str" being exported from the plugin
2. You should be importing "my-unplugin-virtual-id/webpack" in this statement and in `webpack.config.js`.

**Comment by Codpoe** [comment]
- the name of the plugin  is `my-plugin`
- `my-unplugin-virtual-id` is the virtual module which is expected to be resolved in `resolveId` function.
- `import str from 'my-unplugin-virtual-id'` is in the business code , not my configuration.

**Comment by TropicalRaisel** [comment]
Ok I'm understanding now. The variable naming was a bit confusing. The name of the plugin in that configuration is `unplugin-starter`, so that's likely your issue. You should be doing this in your Webpack config:
```javascript
module.exports = {
  entry: './src/index.js',
  plugins: [
    require('unplugin-starter/webpack').default(),
  ],
}
```

**Comment by Codpoe** [comment]
Yeah, this is my current configuration, but the `resolveId` function doesn't work in webpack project as expected, and it throw an error which indicates the virtual module is not found.

**Comment by TropicalRaisel** [comment]
Try deleting your `load` section and see if it works then. My suspicion is this error is thrown there and not in `resolveId`.

**Comment by Codpoe** [comment]
I tried to do what you said, but it still didn’t work

**Comment by danielroe** [comment]
I'm getting this as well. In case it's helpful, here are three reproductions:

* unplugin repro: https://github.com/Akryum/unplugin-icons-issue-94/tree/unplugin - see https://github.com/antfu/unplugin-icons/issues/94#issuecomment-940966103
* unplugin-icons + nuxt repro: https://stackblitz.com/edit/nuxt-starter-pbfj9k
* unplugin + nuxt repro: https://stackblitz.com/edit/nuxt-starter-repdpw

**Comment by yunsii** [comment]
Unplugin `resolveId` seem still not work in webpack on Windows OS.

**Comment by yunsii** [comment]
> Unplugin `resolveId` with virtual modules seem still not work in webpack on Windows OS.

It works today, but i don't konw what happens 😦
